bindings/csharp/Device.cs

```diff
@@ -101,7 +102,43 @@ public override void write(string str)
             }
         }
 
-        private Context ctx;
+        private class DeviceBufferAttr : Attr
+        {
+            private IntPtr dev;
+
+            [DllImport("libiio.dll", CallingConvention = CallingConvention.Cdecl)]
+            private static extern int iio_device_buffer_attr_read(IntPtr dev, [In] string name, [Out()] StringBuilder val, uint len);
+            
+            [DllImport("libiio.dll", CallingConvention = CallingConvention.Cdecl)]
+            private static extern int iio_device_buffer_attr_write(IntPtr dev, [In] string name, [In()] string val);
+
+            public DeviceBufferAttr(IntPtr dev, string name) : base(name)
+            {
+                this.dev = dev;
+            }
+
+            public override string read()
+            {
+                StringBuilder builder = new StringBuilder(1024);
```

Comment:
Why 1024?

Reply:
It was the same as in the DeviceDebugAttr.read() method, which was already implemented. Should I change these values?

Reply:
Some attributes might be longer than this. Have a look at how this is handled in iio_info.

Reply:
I think we determined the min should be a page size (4096).

Reply:
I have changed the size to the value of BUF_SIZE (16384).